src/otx/algo/classification/torchvision_model.py

```diff
@@ -0,0 +1,250 @@
+    """
+
+    def __init__(
+        self,
+        backbone: Literal[
+            "resnet50",
+            "efficientnet_b0",
+            "efficientnet_b1",
+            "efficientnet_b3",
+            "efficientnet_b4",
+            "efficientnet_v2_l",
+            "mobilenet_v3_small",
+        ],
+        num_classes: int,
+        loss: nn.Module | None = None,
+    ) -> None:
+        self.backbone = backbone
+        self.loss = loss
+
+        super().__init__(num_classes=num_classes)
+
+    def _create_model(self) -> nn.Module:
+        return TVModelWithLossComputation(
+            backbone=self.backbone,
+            num_classes=self.num_classes,
+            loss=self.loss,
+        )
+
+    def _customize_inputs(self, inputs: MulticlassClsBatchDataEntity) -> dict[str, Any]:
+        if isinstance(inputs.images, list):
+            images = tv_tensors.wrap(torch.stack(inputs.images, dim=0).to(dtype=torch.float32), like=inputs.images[0])
+        else:
+            images = inputs.images.to(dtype=torch.float32)
+        return {
+            "images": images,
+            "labels": torch.cat(inputs.labels, dim=0),
+            "mode": "loss" if self.training else "predict",
+        }
+
+    def _customize_outputs(
+        self,
+        outputs: Any,  # noqa: ANN401
+        inputs: MulticlassClsBatchDataEntity,
+    ) -> MulticlassClsBatchPredEntity | MulticlassClsBatchPredEntityWithXAI | OTXBatchLossEntity:
+        if self.training:
+            return OTXBatchLossEntity(loss=outputs)
+
+        # To list, batch-wise
+        logits = outputs if isinstance(outputs, torch.Tensor) else outputs["logits"]
+        scores = torch.unbind(logits, 0)
+        preds = logits.argmax(-1, keepdim=True).unbind(0)
+
+        if self.explain_mode:
+            if not isinstance(outputs, dict) or "saliency_map" not in outputs:
+                msg = "No saliency maps in the model output."
+                raise ValueError(msg)
+
+            saliency_maps = outputs["saliency_map"].detach().cpu().numpy()
+
+            return MulticlassClsBatchPredEntityWithXAI(
+                batch_size=len(preds),
+                images=inputs.images,
+                imgs_info=inputs.imgs_info,
+                scores=scores,
+                labels=preds,
+                saliency_maps=list(saliency_maps),
+                feature_vectors=[],
+            )
+
+        return MulticlassClsBatchPredEntity(
+            batch_size=inputs.batch_size,
+            images=inputs.images,
+            imgs_info=inputs.imgs_info,
+            scores=scores,
+            labels=preds,
+        )
+
+    @property
+    def _export_parameters(self) -> dict[str, Any]:
+        """Defines parameters required to export a particular model implementation."""
+        export_params: dict[str, Any] = {}
+        export_params["input_size"] = (1, 3, 224, 224)
+        export_params["output_names"] = ["logits", "saliency_map"] if self.explain_mode else None
```

Comment:
`export_params["output_names"]` can be moved to `ExplainableOTXClsModel`, since that class owns the explain forward.